lib/utils/file_transfer.rb

```diff
@@ -0,0 +1,76 @@
+require "services"
+require "utils/file_transfer_error"
+
+module Utils
+  # Use rclone system calls to transfer files between
+  # local and remote storage (and vice versa).
+  class FileTransfer
+    def initialize
+      # Check that config is set...
+      if Settings.rclone_config_path.nil?
+        raise "Settings.rclone_config_path missing from Settings"
+      end
+      # ... and points to an actual file.
+      unless File.exist?(Settings.rclone_config_path)
+        raise "Settings.rclone_config_path points to " \
+              "#{Settings.rclone_config_path}, which does not exist"
+      end
+    end
+
+    # Really just an alias for transfer
+    def upload(local_file, remote_dir)
+      transfer(local_file, remote_dir)
+    end
+
+    # Really just an alias for transfer
+    def download(remote_file, local_dir)
+      transfer(remote_file, local_dir)
+    end
+
+    def exists?(dir)
+      make_call("#{call_prefix} ls \"#{dir}\"")
+    end
+
+    def mkdir(dir)
+      make_call("#{call_prefix} mkdir \"#{dir}\"")
+    end
+
+    # Return parsed JSON ls output
+    def ls_remote_dir(remote_dir)
```

Comment:
`exists?` and `mkdir` are both (I assume) primarily for remote directories; consider calling this just `ls` rather than `ls_remote_dir` for parallelism?

Reply:
renamed it lsjson, which is what it does